filebeat/module/elasticsearch/deprecation/ingest/pipeline.json

```diff
@@ -43,7 +43,8 @@
         {
             "date": {
                 "if": "ctx.event.timezone != null",
-                "field": "@timestamp",
```

Comment:
@pragkent Thanks for looking at these modules too, but let's focus first on the system module. For the elasticsearch one I think we will also need to change the formats, because it is actually like `ISO8601`, but without timezone.

Reply:
@jsoriano yes, you are correct. we need to define a new format, or the timezone is wrong. I'll fix it later.

Reply:
I grabbed some elasticsearch 7.3 logs from our server, it seems that ISO8601 format was used (timezone included). So, use ISO8601 format for parsing is actually right. 

And logs from elasticsearch 6.x does not have timezone part, as you said.

So I added a format for parsing elasticsearch 6.x logs.

Could you please double-check that?

Part of the logs is listed as follow:

```log
{"type": "deprecation", "timestamp": "2019-08-20T05:08:27,756+0000", "level": "WARN", "component": "o.e.d.c.s.Settings", "cluster.name": "hulk", "node.name": "hulk-es-22x9pqqc4q",  "message": "[discovery.zen.hosts_provider] setting was deprecated in Elasticsearch and will be removed in a future release! See the breaking changes documentation for the next major version."  }

{"type": "server", "timestamp": "2019-08-22T20:52:56,131+0000", "level": "INFO", "component": "o.e.m.j.JvmGcMonitorService", "cluster.name": "hulk", "node.name": "hulk-es-22x9pqqc4q", "cluster.uuid": "3pYlYTCPT4iLMqHmeq7Yew", "node.id": "UFfAvAUkTIuOlUbMgT9x0A",  "message": "[gc][229011] overhead, spent [293ms] collecting in the last [1s]"  }
```

Reply:
Looks good, but I think we could remove the `ISO8601` format.

Reply:
Or do something like what I proposed for `ISO8601` in the system module.

Reply:
Hmm, I think we should not remove ISO8601 for elasticsearch module.  For example, if we have elasticsearch 7.3, and have event.timezone set, then the parsing will failed without this ISO8601 format.

Reply:
I am trying and this seems correct in 7.3 for JSON logs, but not for plain text logs. We may need to move date parsing to the specific pipeline (there are three pipelines, a common one, and another two for the specific needings of plain text and json formats).
Also I think that the format for timestamps without timezone should be `yyyy-MM-dd'T'HH:mm:ss,SSS`, as they contain milliseconds.

Reply:
You are right. I just changed the format to `yyyy-MM-dd'T'HH:mm:ss,SSS`.  When the format is `yyyy-MM-dd'T'HH:mm:ss`, the date processor will always choose ISO8601 format for parsing, because the time in log files contain milliseconds part.

Is there any other reasons which caused plain text logs parsing failure that I missed out?